File: www/cgi-bin/env_dump.py
#!/usr/bin/env python3
import os
import sys

body = ""
try:
    length = int(os.environ.get("CONTENT_LENGTH", "0") or "0")
    if length > 0:
        body = sys.stdin.read(length)
except (ValueError, Exception):
    pass

print("Content-Type: text/plain")
print()
print("=== ENV DUMP (python) ===")
for key in sorted(os.environ.keys()):
    print(f"{key}={os.environ[key]}")
print()
print("=== BODY ===")
print(body)

File: www/cgi-bin/env_dump.pl
#!/usr/bin/perl
use strict;
use warnings;

my $body = "";
my $len = $ENV{"CONTENT_LENGTH"} || 0;
if ($len > 0) {
    read(STDIN, $body, $len);
}

print "Content-Type: text/plain\r\n\r\n";
print "=== ENV DUMP (perl) ===\n";
foreach my $key (sort keys %ENV) {
    print "$key=$ENV{$key}\n";
}
print "\n=== BODY ===\n";
print "$body\n";
